core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceTest.java

```diff
@@ -672,6 +708,69 @@ public String headerWithoutValue(RequestContext ctx,
         }
     }
 
+    static class MyAnnotatedService12 {
+
+        @Get("/httpResponse")
+        public HttpResponse httpResponse(RequestContext ctx) {
+            validateContext(ctx);
+            return HttpResponse.of(HttpStatus.OK);
+        }
+
+        @Get("/aggregatedHttpResponse")
+        public AggregatedHttpResponse aggregatedHttpResponse(RequestContext ctx) {
+            validateContext(ctx);
+            return AggregatedHttpResponse.of(HttpStatus.OK);
+        }
+
+        @Get("/jsonNode")
+        public JsonNode jsonNode(RequestContext ctx) {
+            validateContext(ctx);
+            return TextNode.valueOf("Armeria");
+        }
+
+        @Get("/completionStage")
+        public CompletionStage<String> completionStage(RequestContext ctx) {
+            validateContext(ctx);
+            return CompletableFuture.supplyAsync(() -> "Armeria");
+        }
+    }
+
+    static class MyAnnotatedService13 {
+
+        @Get("/httpResponse")
+        @Blocking
+        public HttpResponse httpResponse(RequestContext ctx) {
+            validateContext(ctx);
+            return HttpResponse.of(HttpStatus.OK);
+        }
+
+        @Get("/aggregatedHttpResponse")
+        @Blocking
+        public AggregatedHttpResponse aggregatedHttpResponse(RequestContext ctx) {
+            validateContext(ctx);
+            return AggregatedHttpResponse.of(HttpStatus.OK);
+        }
+
+        @Get("/jsonNode")
+        @Blocking
+        public JsonNode jsonNode(RequestContext ctx) {
+            validateContext(ctx);
+            return TextNode.valueOf("Armeria");
+        }
+
+        @Get("/completionStage")
+        @Blocking
+        public CompletionStage<String> completionStage(RequestContext ctx) {
+            validateContext(ctx);
+            return CompletableFuture.supplyAsync(() -> "Armeria");
+        }
+    }
+
+    @BeforeEach
+    void clear() {
+        blockingCount.set(0);
+    }
```

Comment:
How about moving this logic up near the `blockingCount` initialization? or remove `final` and create `new AtomicInteger` before each test?